src/code/z_frame_advance.c

```diff
@@ -0,0 +1,25 @@
+#include "global.h"
+
+void FrameAdvance_Init(FrameAdvanceContext* frameAdvCtx) {
+    frameAdvCtx->timer = 0;
+    frameAdvCtx->enabled = false;
+}
+
+/**
+ * Frame advance allows you to advance through the game one frame at a time on command.
+ * To advance a frame, hold Z and press R on the specified controller.
+ * Holding Z and R will advance a frame every half second.
+ *
+ * This function returns true when frame advance is not active (game will run normally)
+ *
+ * Note: There is no way to enable it
+ */
```

Comment:
Since this comment really applies to the whole system, it seems better to put it at the top of the file in a file header.